### PR TITLE
Feature/crashes check

### PR DESCRIPTION
**Pull request by gladky**
Implementation of #134

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -216,6 +225,12 @@
 		</dependency>
 
 
+		<dependency>
+			<groupId>org.javassist</groupId>
+			<artifactId>javassist</artifactId>
+			<version>3.21.0-GA</version>
+		</dependency>
+
```
Comment (andreh12) [inline pom.xml]: what do we need this for ? (do we need it in the future ? -- I see it seems to be excluded above)

File: src/main/java/rcms/utilities/daqexpert/reasoning/logic/failures/CmsswCrashes.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            return true;
+
+
+    }
+
+    @Override
+    public void parametrize(Properties properties) {
+        this.crashesCountThreshold = FailFastParameterReader.getIntegerParameter(properties, Setting.EXPERT_CMSSW_CRASHES_THRESHOLD, this.getClass());
+        this.slidingWindowPeriodInSeconds = FailFastParameterReader.getIntegerParameter(properties, Setting.EXPERT_CMSSW_CRASHES_TIME_WINDOW, this.getClass());
+        this.holdoffPeriodInSeconds = FailFastParameterReader.getIntegerParameter(properties, Setting.EXPERT_CMSSW_CRASHES_HOLDOFF, this.getClass());
```
Comment (andreh12) [inline src/main/java/rcms/utilities/daqexpert/reasoning/logic/failures/CmsswCrashes.java]: previously time parameters (e.g. `expert.logic.continoussofterror.threshold.period`) were specified in milliseconds while these seem to be in seconds. Should we globally move to seconds instead (this is more human friendly in the configuration files) ?

File: pom.xml (hunk 1 above)
Comment (gladky) [reply pom.xml]: It's related with javaassist's problem with java 8. The problem is discussed in [this SO post](https://stackoverflow.com/questions/24281235/error-creating-entitymanagerfactory-due-to-error-tying-to-scan-jar-file).

File: src/main/java/rcms/utilities/daqexpert/reasoning/logic/failures/CmsswCrashes.java (hunk 2 above)
Comment (gladky) [reply src/main/java/rcms/utilities/daqexpert/reasoning/logic/failures/CmsswCrashes.java]: That's valid point, the seconds may be more readable. Let's convert everything to seconds after integrating all of the deadtime analalysis related branches. We could actually also review the properties names convention.

Comment (andreh12) [reply src/main/java/rcms/utilities/daqexpert/reasoning/logic/failures/CmsswCrashes.java]: ok, opened #161 to keep track of this for the other modules which use milliseconds.

File: pom.xml (hunk 1 above)
Comment (andreh12) [reply pom.xml]: ok -- why do we need javassist ? For hibernate ?

**Comment by gladky** [comment]
remove the holdoff

File: pom.xml (hunk 1 above)
Comment (gladky) [reply pom.xml]: Yes, without javassist hibernate throws an exception in our integration tests:

```
Tests run: 3, Failures: 0, Errors: 1, Skipped: 2, Time elapsed: 1.151 sec <<< FAILURE!
test(rcms.utilities.daqexpert.processing.JobManagerIT)  Time elapsed: 0.035 sec  <<< ERROR!
java.lang.NoClassDefFoundError: javassist/bytecode/ClassFile
        at org.hibernate.ejb.packaging.AbstractJarVisitor.checkAnnotationMatching(AbstractJarVisitor.java:243)
        at org.hibernate.ejb.packaging.AbstractJarVisitor.executeJavaElementFilter(AbstractJarVisitor.java:209)
        at org.hibernate.ejb.packaging.AbstractJarVisitor.addElement(AbstractJarVisitor.java:170)
```

Comment (andreh12) [reply pom.xml]: ok, thanks for the clarification !

**Pull request merged**